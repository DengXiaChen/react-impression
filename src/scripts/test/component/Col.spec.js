import React from 'react';
import { shallow, mount } from 'enzyme';
import { Row, Col } from '../../components/impression';

const setup = () => {
    const props = {
    };

    const wrapper = shallow(
        <Row>
            <Col>4</Col>
            <Col>4</Col>
            <Col>4</Col>
        </Row>
    );

    const wrapper1 = mount(
        <Row>
            <Col>4</Col>
            <Col>4</Col>
            <Col>4</Col>
        </Row>
    );

    return {
        props,
        wrapper,
        wrapper1,
    };
};

describe('Row Col', () => {
    const { wrapper1, wrapper } = setup();

    it('render', () => {
        expect(wrapper1.find('.row').exists()).toBe(true);
        expect(wrapper1.find('.col-xs-4')).toHaveLength(3);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
