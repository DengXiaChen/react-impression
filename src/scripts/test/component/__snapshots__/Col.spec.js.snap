// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Col ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="row"
>
    <Col
        col={4}
    >
        4
    </Col>
    <Col
        col={4}
    >
        4
    </Col>
    <Col
        col={4}
    >
        4
    </Col>
</div>,
  "nodes": Array [
    <div
      className="row"
>
      <Col
            col={4}
      >
            4
      </Col>
      <Col
            col={4}
      >
            4
      </Col>
      <Col
            col={4}
      >
            4
      </Col>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Row>
        <Col>
                4
        </Col>
        <Col>
                4
        </Col>
        <Col>
                4
        </Col>
</Row>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Row {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <Col>
              4
</Col>,
            <Col>
              4
</Col>,
            <Col>
              4
</Col>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="row"
>
          <Col
                    col={4}
          >
                    4
          </Col>
          <Col
                    col={4}
          >
                    4
          </Col>
          <Col
                    col={4}
          >
                    4
          </Col>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="row"
>
          <Col
                    col={4}
          >
                    4
          </Col>
          <Col
                    col={4}
          >
                    4
          </Col>
          <Col
                    col={4}
          >
                    4
          </Col>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Row>
    <Col>
        4
    </Col>
    <Col>
        4
    </Col>
    <Col>
        4
    </Col>
</Row>,
}
`;
